home/views.py

```diff
@@ -14,17 +16,35 @@ def landingpage(request):
 
 def displayQuestion(request, **kwargs):
     question = Question.objects.all().get(id=kwargs['pk'])
-    sortAnsBy = request.GET["sortanswersby"] if 'sortanswersby' in request.GET else ''
+    sort_answer_by = request.GET.get("sortanswersby", "")
     context = {
         "question": question,
-        "answers": question.get_answers_feed(sortAnsBy),
+        "answers_structure": question.create_answers_structure(sort_answer_by, request.profile),
         "answersCount": question.answer_set.count(),
         "tags": question.tags.values(),
-        "title": question.get_question_title()
+        "title": question.get_question_title(),
+        "is_user_logged_in": request.user.is_authenticated
     }
     return render(request, 'home/question_detail.html', context)
 
 
+def thumbs(request):
+    if request.user.is_authenticated:
+        button = str(request.POST.get('button'))
+        (thumb_type, answer_id) = button.split(' ')
```

Comment:
this is making too many assumption about the button string in the form post data - who said its even splittable? not to mention containing the right "thumb" string? it could be total garbage...

Reply:
fixed - not using form post any more